### PR TITLE
Non-implied mode for model cores

### PR DESCRIPTION
**Pull request by ajreynol**
This introduces a new "non-implied" mode for model cores.

@barrettcw, you may be interested in this PR.

### CODE REVIEW

File: test/regress/regress0/model-core.smt2 [hunk 1]
```diff
@@ -1,4 +1,4 @@
-; COMMAND-LINE: --produce-model-cores
+; COMMAND-LINE: --produce-models --model-cores=simple
```
Comment (yoni206) [inline test/regress/regress0/model-core.smt2]: Perhaps we should test this both for simple and for non-implied modes?

```suggestion
; COMMAND-LINE: --produce-models --model-cores=simple
;COMMAND-LINE: --produce-models --model-core=non-implied
```

Comment (yoni206) [inline test/regress/regress0/model-core.smt2]: What do you think about adding Alessandro's example as another test case for both modes (maybe in regress1)?

Comment (yoni206) [inline test/regress/regress0/model-core.smt2]: Is there a test case where the model core is predictable and robust, in which we can use ;EXPECT in order to verify that the model core is what we expected it to be?

File: src/theory/subs_minimize.cpp [hunk 2]
```diff
@@ -124,8 +262,6 @@ bool SubstitutionMinimize::find(Node n,
   Trace("subs-min") << "--- Compute relevant variables..." << std::endl;
   std::unordered_set<Node, NodeHashFunction> rlvFv;
   // only variables that occur in assertions are relevant
```
Comment (yoni206) [inline src/theory/subs_minimize.cpp]: Does this comment still belong here?

Comment (ajreynol) [reply src/theory/subs_minimize.cpp]: Yes, relevant here means "possibly required or implied".

File: test/regress/regress0/model-core.smt2 (hunk 1 above)
Comment (4tXJ7f) [reply test/regress/regress0/model-core.smt2]: Is it possible to implement something similar to `--check-models` for model cores (maybe out-of-scope for this PR)?

**Pull request merged**